server/specs/db.test.js

```diff
@@ -0,0 +1,119 @@
+const db = require('../db.js');
+
+
+describe('DB', () => {
+  
+  test('Should return an array when asked for similar listings', (done) => {
+    db.getSimilarListings(1, (err, result) => {
+      expect(Array.isArray(result)).toBe(true);
+      done();
+    });
+  });
+
+  test('Should return similar listing for given listing id', (done) => {
+    db.getSimilarListings(1, (err, result) => {
+      if (err) {
+        console.log(err);
+      } else {
+        expect(result[0].keywords.includes('home')).toBe(true);
+        done();
+      }
+     
+    });
+  });
+
+});
+
```

Comment:
@jennqiao Everything looks good for me.
I have a doubt here, what is the result if there is no keywords matching? how will SimilarListings be displayed?

Reply:
That's a great question. If there are no keywords matching (fewer than 3), it looks for price similarities (anything within $100 range of the price) and displays those instead.